pkg/gadget-context/gadget-context.go

```diff
@@ -142,6 +175,126 @@ func (c *GadgetContext) GadgetInfo() *runTypes.GadgetInfo {
 	return c.gadgetInfo
 }
 
+func (c *GadgetContext) RegisterDataSource(t datasource.Type, name string) (datasource.DataSource, error) {
+	ds := datasource.New(t, name)
+	c.dataSources[name] = ds
+	return ds, nil
+}
+
+func (c *GadgetContext) GetDataSources() map[string]datasource.DataSource {
+	res := make(map[string]datasource.DataSource)
+	for k, v := range c.dataSources {
+		res[k] = v
+	}
+	return res
```

Comment:
I like to return copies in those cases as handing over the original could also mess up the "internals".
But yeah, `maps.Clone()` would work since Go 1.21 - I'll change it 😄